python/pylibcugraph/pylibcugraph/tests/conftest.py

```diff
@@ -11,15 +11,196 @@
+    weights=cp.asarray([0.1, 2.1, 1.1, 5.1, 3.1, 4.1, 7.2, 3.2],
+                       dtype=np.float32),
+    name="Simple_2"
+    )
+
+
+# The objects in these lists must have a "name" attr, since fixtures will
+# access that to pass to tests, which then may use the name to associate to
+# expected test results. The name attr is also used for the pytest test ID.
+valid_datasets = [utils.RAPIDS_DATASET_ROOT_DIR_PATH/"karate.csv",
+                  utils.RAPIDS_DATASET_ROOT_DIR_PATH/"dolphins.csv",
+                  Simple_1,
+                  Simple_2,
+                  ]
+all_datasets = valid_datasets + \
+               [InvalidNumWeights_1,
+                InvalidNumVerts_1,
+                ]
+
+
+# =============================================================================
+# Helper functions
+# =============================================================================
+def get_graph_data_for_dataset(ds, ds_name):
+    """
+    Given an object representing either a path to a dataset on disk, or an
+    object containing raw data, return a series of arrays that can be used to
+    construct a graph object. The final value is a bool used to indicate if the
+    data is valid or not (invalid to test error handling).
+    """
+    if isinstance(ds, COOTestGraphDeviceData):
+        device_srcs = ds.srcs
+        device_dsts = ds.dsts
+        device_weights = ds.weights
+        is_valid = ds.is_valid
+    else:
+        pdf = pd.read_csv(ds,
+                          delimiter=" ", header=None,
+                          names=["0", "1", "weight"],
+                          dtype={"0": "int32", "1": "int32",
+                                 "weight": "float32"},
+                          )
+        device_srcs = cp.asarray(pdf["0"].to_numpy(), dtype=np.int32)
+        device_dsts = cp.asarray(pdf["1"].to_numpy(), dtype=np.int32)
+        device_weights = cp.asarray(pdf["weight"].to_numpy(), dtype=np.float32)
+        # Assume all datasets on disk are valid
+        is_valid = True
+
+    return (device_srcs, device_dsts, device_weights, ds_name, is_valid)
+
+
+def create_SGGraph(device_srcs,
+                   device_dsts,
+                   device_weights,
+                   transposed=False):
+    """
+    Creates and returns a SGGraph instance and the corresponding ResourceHandle
+    using the parameters passed in.
+    """
+    from pylibcugraph.experimental import (SGGraph,
+                                           ResourceHandle,
+                                           GraphProperties,
+                                           )
+    resource_handle = ResourceHandle()
+    graph_props = GraphProperties(is_symmetric=False, is_multigraph=False)
+
+    g = SGGraph(resource_handle,
+                graph_props,
+                device_srcs,
+                device_dsts,
+                device_weights,
+                store_transposed=transposed,
+                renumber=False,
```

Comment:
So, are we disabling renumbering for SG? This has negative performance implications.

Reply:
It's not disabled in the API; the code here is just for creating graphs for tests.  I'll add a FIXME to add coverage for both renumbering enabled and disabled in the next commit.